packages/vite/src/node/plugins/oxc.ts

```diff
@@ -176,8 +195,19 @@ export function oxcPlugin(config: ResolvedConfig): Plugin {
       // recycle serve to avoid preventing Node self-exit (#6815)
       setServer(null)
     },
-    async transform(code, id) {
+    async transform(code, id, options) {
       if (filter(id) || filter(cleanUrl(id))) {
+        // disable refresh at ssr
+        if (options?.ssr && oxcTransformOptions.jsx?.refresh) {
```

Comment:
```suggestion
    async transform(code, id) {
      if (filter(id) || filter(cleanUrl(id))) {
        // disable refresh at ssr
        if (this.environment.config.consumer === 'server' && oxcTransformOptions.jsx?.refresh) {
```
The SSR flag is now replace with this, since the env API introduction.